include/ginkgo/core/base/lin_op.hpp

```diff
@@ -585,6 +593,27 @@ class Preconditionable {
 };
 
 
+/**
+ * The diagonal of a LinOp implementing this interface can be extracted.
```

Comment:
Could you add the information of diag here?
`extract_diagonal` extracts the elements whose col and row index are the same. It returns the `min(nrows, ncols) x 1` dense matrix.